src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerMetricsDiagnosticListener.cs

```diff
@@ -104,4 +107,36 @@ public override void OnEventWritten(string name, object payload)
             }
         }
     }
+
+    // The AOT-annotation DynamicallyAccessedMembers(https://learn.microsoft.com/dotnet/api/system.diagnostics.codeanalysis.dynamicallyaccessedmembersattribute?view=net-7.0)
+    // in Systm.Net.Http library ensures that top-level properties on the payload object are always preserved.
+    // see https://github.com/dotnet/runtime/blob/f9246538e3d49b90b0e9128d7b1defef57cd6911/src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L325
+#if NET6_0_OR_GREATER
+    [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = "The event source guarantees that top-level properties are preserved")]
+#endif
+    private bool TryFetchResponse(object payload, out HttpResponseMessage response)
```

Comment:
This approach definitely works so just a personal opinion: I would consider using local functions (it doesn't pollute the class) if it has only one callsite.

Reply:
Alternatively you could have just one generic helper (generic on the TPropertyType just like property fetcher) and give it the property fetcher and payload. If it's private to this class it's pretty clearly scoped to just HTTP messages. But it would have the risk of somebody just copying it elsewhere without thinking about it.

Reply:
I would probably go with local static functions (to avoid closure state allocation) - but it's similar to this approach - so just a matter of preference.

Reply:
Good idea! I've updated the PropertyFetcher helper methods in this PR to be local static functions. Thank you.